server/controllers/Users.js

```diff
@@ -20,6 +20,12 @@ usersController.post('/registration', async (req, res) => {
     await UserService.registerUser(userData)
     res.json(userData)
 })
+usersController.get('/check-username/{username}', async (req, res) => {
```

Comment:
```suggestion
usersController.get('/username/{username}', async (req, res) => {
```
You might just proceed with /username/{username}. Implicitly, a GET request checks if a given resource is present or not

Reply:
Avoid using verbs within an API path. Refer to this: https://restfulapi.net/resource-naming/

Reply:
this is out of scope from team page